### PR TITLE
rootless: fix CONTAINERD_ROOTLESS_ROOTLESSKIT_PORT_DRIVER=slirp4netns

### PR DESCRIPTION
**Pull request by AkihiroSuda**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,10 +53,27 @@ jobs:
...
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 1
+    - name: "Prepare (network driver=slirp4netns, port driver=builtin)"
       run: DOCKER_BUILDKIT=1 docker build -t test-rootless --target test-rootless --build-arg CONTAINERD_VERSION=${CONTAINERD_VERSION} .
-    - name: "Test (rootless)"
+    - name: "Test    (network driver=slirp4netns, port driver=builtin)"
       run: docker run -t --rm --privileged --tmpfs /tmp:exec test-rootless
+    - name: "Prepare (network driver=slirp4netns, port driver=slirp4netns)"
+      run: DOCKER_BUILDKIT=1 docker build -t test-rootless --target test-rootless-port-slirp4netns --build-arg CONTAINERD_VERSION=${CONTAINERD_VERSION} .
```
Comment (ktock) [inline .github/workflows/test.yml]: `-t test-rootless-port-slirp4netns` ?

File: pkg/ocihook/ocihook.go [hunk 2]
```diff
@@ -194,17 +202,38 @@ func getCNINamespaceOpts(opts *handlerOpts) ([]cni.NamespaceOpts, error) {
...
-			if hostIP := net.ParseIP(p.HostIP); hostIP != nil {
+			if hostIP := net.ParseIP(p.HostIP); hostIP != nil && !hostIP.IsUnspecified() {
 				// loopback address is always bindable in the child namespace, but other addresses are unlikely.
 				if !hostIP.IsLoopback() {
-					p.HostIP = "127.0.0.1"
+					if childIP != nil && childIP.Equal(hostIP) {
+						// this is fine
+					} else {
+						if portDriverDisallowsLoopbackChildIP {
+							p.HostIP = childIP.String()
```
Comment (ktock) [inline pkg/ocihook/ocihook.go:230]: Is it OK that `childIP == nil` enters this case?

File: pkg/ocihook/ocihook.go [hunk 3]
```diff
@@ -194,17 +202,38 @@ func getCNINamespaceOpts(opts *handlerOpts) ([]cni.NamespaceOpts, error) {
...
 		// For rootless, we need to modify the hostIP that is not bindable in the child namespace.
 		// https: //github.com/containerd/nerdctl/issues/88
 		//
 		// We must NOT modify opts.ports here, because we use the unmodified opts.ports for
 		// interaction with RootlessKit API.
 		ports := make([]cni.PortMapping, len(opts.ports))
 		for i, p := range opts.ports {
-			if hostIP := net.ParseIP(p.HostIP); hostIP != nil {
+			if hostIP := net.ParseIP(p.HostIP); hostIP != nil && !hostIP.IsUnspecified() {
 				// loopback address is always bindable in the child namespace, but other addresses are unlikely.
```
Comment (ktock) [inline pkg/ocihook/ocihook.go:224]: nit: 
not `always bindable`? (i.e. loopback is not bindable when `portDriverDisallowsLoopbackChildIP == true`?)

File: .github/workflows/test.yml (hunk 1 above)
Comment (ktock) [reply .github/workflows/test.yml]: nevermind. this is outdated review.

File: pkg/ocihook/ocihook.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/ocihook/ocihook.go:230]: Can’t be nil when portDriverDisallowsLoopbackChildIP is true

File: pkg/ocihook/ocihook.go (hunk 3 above)
Comment (AkihiroSuda) [reply pkg/ocihook/ocihook.go:224]: Bindable but not forwardable

**Pull request merged**